app/src/test/java/rick_and_morty/ui/character_details/CharacterShortDetailsViewModelTest.kt

```diff
@@ -0,0 +1,123 @@
+import rick_and_morty.rules.CoroutineTestRule
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class CharacterShortDetailsViewModelTest {
+
+
+    @get:Rule
+    val coroutineTestRule = CoroutineTestRule()
+
+    private val originDto = OriginDto(
+        name = "Earth (C-137)",
+        url = "https://rickandmortyapi.com/api/location/1"
+    )
+
+    private val locationDto = LocationDto(
+        name = "Citadel of Ricks",
+        url = "https://rickandmortyapi.com/api/location/3"
+    )
+
+    private val characterResultsDto =
+        CharacterResultsDto(
+            id = 1,
+            name = "Chicko Micko",
+            status = "Alive",
+            species = "Human",
+            type = "",
+            gender = "Male",
+            originDto = originDto,
+            locationDto = locationDto,
+            image = "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
+            episode = listOf(
+                "https://rickandmortyapi.com/api/episode/1",
+                "https://rickandmortyapi.com/api/episode/2",
+                "https://rickandmortyapi.com/api/episode/3",
+                "https://rickandmortyapi.com/api/episode/4",
+                "https://rickandmortyapi.com/api/episode/5"
+            ),
+            url = "https://rickandmortyapi.com/api/character/1",
+            created = "2017-11-04T18:48:46.250Z"
+        )
+
+    private val expectedCharacterShortDetailsLists = CharacterDetails(
+        characterResultsDto.image,
+        listOf(
+            CharacterShortDetails("Name", "Chicko Micko"),
+            CharacterShortDetails("Last known location", "Citadel of Ricks"),
+            CharacterShortDetails("Species", "Human"),
+            CharacterShortDetails("Created", "2017-11-04"),
+            CharacterShortDetails("Gender", "Male"),
+            CharacterShortDetails("Origin", "Earth (C-137)"),
+            CharacterShortDetails("Status", "Alive"))
+    )
+    private val expectedEmptyCharacterShortDetailsLists =
+        CharacterDetails(image="", characterShortDetailsList= emptyList())
+
+    private val savedStateHandle = SavedStateHandle (mapOf("characterID" to 1))
+
+    private val characterRepository: CharacterRepository = mock {
+        onBlocking { it.getCharacterDetails(1) } doReturn (characterResultsDto)
+    }
+
+    private val classToTest by lazy {
+        CharacterDetailsViewModel(
+            characterRepository,
+            savedStateHandle
+        )
+    }
+    @Test
+    fun `return Empty When Init ViewModel`() = runTest {
+        given(characterRepository.getCharacterDetails(any())).willReturn(null)
+        assertThat(classToTest.characterDetails.value.characterResultDetails).isEqualTo(expectedEmptyCharacterShortDetailsLists)
+
+    }
+    @Test
+    fun `return List When GetCharacterDetails Called`() = runTest {
+
+        classToTest.getCharacterDetails(1)
+
+        advanceUntilIdle()
+
+        verify(characterRepository, times(2)).getCharacterDetails(1)
```

Comment:
unless there is some sort of retry functionality, this never happens in the actual code. `getCharacters` is called exactly once, in the `init` block